app/src/main/javascript/bundles/experiment-page/src/ExperimentPageRouter.js

```diff
@@ -24,11 +24,30 @@ const TabCommonPropTypes = {
 }
 
 // What component each tab type should render, coupled to ExperimentController.java
-const tabTypeComponent = {
-  'results' : TSnePlotViewRoute,
-  'experiment-design' : ExperimentDesignRoute,
-  'supplementary-information' : SupplementaryInformationRoute,
-  'downloads' : DownloadsRoute
+let tabTypeComponent = []
+
+function enableExperimentPageTab(tab) {
+  if (tab.type === 'results') {
+    if (Array.isArray(tab.props.ks) && (tab.props.plotTypesAndOptions)) {
+      tabTypeComponent.push({'results': TSnePlotViewRoute})
+      return tab.name;
+    }
+  } else if (tab.type === 'experiment-design') {
+    if (Array.isArray(tab.props.table.data)) {
+      tabTypeComponent.push({'experiment-design': ExperimentDesignRoute})
+      return tab.name;
+    }
+  } else if (tab.type === 'supplementary-information') {
+    if (Array.isArray(tab.props.sections)) {
+      tabTypeComponent.push({'supplementary-information': SupplementaryInformationRoute})
+      return tab.name;
+    }
+  } else {
```

Comment:
What happens is we are going to have a new `tab.type`?
With this current code it is going to be a `download` tab.
Could you please think it over if that's exactly we want here or something else?
Thanks

Reply:
I got what you mean, I will work on this and let you know.